apps/studio/src/server/modules/user/user.service.ts

```diff
@@ -154,3 +192,108 @@ export const getUsersQuery = ({ siteId, adminType }: GetUsersQueryProps) => {
       "ActiveResourcePermission.userId",
     )
 }
+
+interface DeleteUserPermissionProps {
+  byUserId: string
+  userId: string
+  siteId: number
+}
+
+export const deleteUserPermission = async ({
+  byUserId,
+  userId,
+  siteId,
+}: DeleteUserPermissionProps) => {
+  // Putting outside the tx to reduce unnecessary extended DB locks
+  const byUser = await db
+    .selectFrom("User")
+    .where("id", "=", byUserId)
+    .selectAll()
+    .executeTakeFirstOrThrow()
+
+  await db.transaction().execute(async (tx) => {
+    const userPermissionToDelete = await tx
+      .selectFrom("ResourcePermission")
+      .where("userId", "=", userId)
+      .where("siteId", "=", siteId)
+      .where("deletedAt", "is", null)
+      .selectAll()
+      .executeTakeFirst()
+
+    if (!userPermissionToDelete) {
+      throw new TRPCError({
+        code: "NOT_FOUND",
+        message: "User permissions not found",
+      })
+    }
+
+    const deletedUserPermission = await tx
+      .updateTable("ResourcePermission")
+      .where("id", "=", userPermissionToDelete.id)
+      .set({ deletedAt: new Date() })
+      .returningAll()
+      .executeTakeFirst()
+
+    // NOTE: this is technically impossible because we're executing
+    // inside a tx and this is the same resource which was fetched earlier
+    if (!deletedUserPermission) {
+      throw new TRPCError({
+        code: "INTERNAL_SERVER_ERROR",
+        message:
+          "Something went wrong while attempting to move your resource, please try again later",
+      })
+    }
+
+    await logPermissionEvent(tx, {
+      eventType: "PermissionDelete",
+      by: byUser,
+      delta: { before: userPermissionToDelete, after: deletedUserPermission },
+    })
+  })
+}
+
+interface UpdateUserDetailsProps {
+  byUserId: string
+  userId: string
+  name?: string
+  phone?: string
+}
+
+export const updateUserDetails = async ({
+  byUserId, // just a sanity reminder in case it's called from outside the router
+  userId,
+  name,
+  phone,
+}: UpdateUserDetailsProps) => {
+  if (byUserId !== userId) {
+    throw new TRPCError({
+      code: "FORBIDDEN",
+      message: "You are not allowed to update this user's details",
+    })
+  }
+
+  const updatedUser = await db.transaction().execute(async (tx) => {
```

Comment:
[nit] can we just do the return here because the 2 return blocks below do the same thing hahahahah